### ISSUE TITLE
extra `References` chapter shown in PDF

### ISSUE THREAD

**Issue by sunt05** [sphinx-doc/sphinx#4775]
Subject: extra `References` chapters will show up in `latexpdf` build

### Problem

I put all my references in one chapter so for the `html` build, the references will be shown in one place.
However, for the `latexpdf` build, I notice a chapter `References` with chapter number will show up with no references in it while another standalone chapter `Bibliography` will be attached to the end.

#### Expected results
I'd like the `References` page shown as it is now in the `html` build but the blank `References` chapter can be excluded from the `latexpdf` build.

### Reproducible project / your project
The RTDS webpage:
- http://suews-docs.readthedocs.io/en/latest/references.html

The PDF with extra but blank `References` chapter:
- https://media.readthedocs.org/pdf/suews-docs/latest/suews-docs.pdf

### Remarks
I know this "issue" is more like a question but I got nothing when I researched for a long time on how to achieve that.

**Comment by tk0miya** [comment]
This is caused by the difference between LaTeX builder and others.
I don't think this is a "bug". 
But it would be nice if we can change the builders to same behavior.

Unfortunately, there are no good way to do that at present as far as I know.

**Comment by jfbu** [comment]
Try with this at top of your ``references.rst``.

```rst
.. _refs:

.. raw:: latex

   \begingroup
   \renewcommand\chapter[1]{\endgroup}

References
==========

.. [J11]
   Järvi L, Grimmond CSB & Christen A (2011) The Surface Urban Energy
   and Water Balance Scheme (SUEWS): Evaluation in Los Angeles and
   Vancouver. J. Hydrol. 411, 219-237.

```

If there had not been ``.. _refs:`` I would have used

```
.. raw:: latex

   \phantomsection
   \begingroup
   \renewcommand\chapter[1]{\endgroup}
```

Hope it helps.

**Comment by jfbu** [comment]
wait, sorry, here is correct way for ``:ref:`refs` `` to work fine from elsewhere in your document:

```
.. raw:: latex

   \cleardoublepage
   \begingroup
   \renewcommand\chapter[1]{\endgroup}
   \phantomsection

.. _refs:

References
==========

.. [J11]
   Järvi L, Grimmond CSB & Christen A (2011) The Surface Urban Energy
   and Water Balance Scheme (SUEWS): Evaluation in Los Angeles and
   Vancouver. J. Hydrol. 411, 219-237.
```

Please discard suggestion from my previous post.

**Comment by sunt05** [comment]
Many thanks, @jfbu ! Your method works like a charm! Really glad I finally got this through under your help!

**Comment by sunt05** [comment]
Should we close this issue now? or put another tag to this thread?

**Comment by jfbu** [comment]
@sunt05 glad it came out ok. I think we should keep this issue opened. At least Sphinx documentation should comment on this problem. But it is hard to advise people to use ``raw`` directive officially, as using it is frowned upon practice ;-). Some thinking is needed here...

**Comment by funkyfuture** [comment]
i'm not sure whether i'm dealing with the same issue as the undesired PDF seems not to be hosted anymore, but it seems to be ta case. both, omitting the heading in the references/bibliography file and the raw contents suggested by @jfbu, are removing the extra chapter in my case, using `lualatex`. but it then disappears from the table of contents. does anyone have a hin how to get it back in there?

**Comment by jfbu** [comment]
@funkyfuture with the raw contents method, there still should be a "Bibliography" entry in the table of contents (and bookmarks) of the PDF. If you modify conf.py like this

```
latex_elements = {
    # The paper size ('letterpaper' or 'a4paper').
    #
    # 'papersize': 'letterpaper',

    # The font size ('10pt', '11pt' or '12pt').
    #
    # 'pointsize': '10pt',

    # Additional stuff for the LaTeX preamble.
    #
    'preamble': r'''
\addto\captionsenglish{\renewcommand{\bibname}{References}}
''',

    # Latex figure (float) alignment
    #
    # 'figure_align': 'htbp',
}
```

you will then see "References" in place of "Bibliography" both in bookmarks and actual table of contents, and as title of the actual references chapter in the PDF. Luckily with `lualatex` (which a priori means `polyglossia` rather than LaTeX's `babel`) this works also, in my brief testing.

**Comment by funkyfuture** [comment]
unfortunately this has not the desired effect. fwiw, here's the config for the latex writer i have in place:

```python
language = 'de'

latex_engine = 'lualatex'

latex_use_xindy = False

latex_elements = {
    'extraclassoptions': 'openany,oneside',
    'papersize': 'a4paper',
    'pointsize': '11pt',
    'fontpkg': r'''
        \setmainfont{Latin Modern Sans}
        \setsansfont{Norasi}
        \setmonofont{Hack}
    ''',

    # Additional stuff for the LaTeX preamble.
    'preamble': r'''
\addto\captionsenglish{\renewcommand{\bibname}{References}}
    ''',
}
```

**Comment by jfbu** [comment]
@funkyfuture 

Try with

```
    'preamble': r'''
\addto\captionsgerman{\renewcommand{\bibname}{References}}
    ''',
```

(or perhaps `\captionsngerman` if above fails, but I think polyglossia uses `german` contrarily to babel's `ngerman` --- not tested)

Side remark: if you use `:index:` with German words starting with non-ascii character, then I think indexing will trigger a build failure by lualatex as you have `latex_use_xindy = False`.

**Comment by funkyfuture** [comment]
thank you very much for your effort @jfbu, i had already failingly tried that.

here's the toc part of the resulting LaTeX markup:

```tex
\setcounter{tocdepth}{1}


% This changes the section's title, but doesn't placte it in the toc
\addto\captionsgerman{\renewcommand{\bibname}{Foozifux}}
    

\title{The missing bibliography entry.}
\date{12.09.2018}
\release{}
\author{John Doe}
\newcommand{\sphinxlogo}{\vbox{}}
\renewcommand{\releasename}{}
\makeindex
\begin{document}

\ifdefined\shorthandoff
  \ifnum\catcode`\=\string=\active\shorthandoff{=}\fi
  \ifnum\catcode`\"=\active\shorthandoff{"}\fi
\fi

\maketitle
\sphinxtableofcontents
\phantomsection\label{\detokenize{index::doc}}
```

regarding (not) xindy, luckily i don't have such item. i tried using `xindy`, but that build failed (claimed that files are missing, but they are). didn't want to open an issue as it doesn't trouble me atm, and maybe that's a problem with Ubuntu.

**Comment by funkyfuture** [comment]
ugh, i just noticed that the index and the list of figures also do not appear in the toc.

**Comment by jfbu** [comment]
@funkyfuture is your build on RTD? They [do not do enough runs](https://github.com/rtfd/readthedocs.org/issues/4454) of pdflatex/xelatex/lualatex. Failing that can you share some small project demonstrating the issue.

**Comment by jfbu** [comment]
@funkyfuture Direct usage of `xindy` is not the recommended way. Rather use the `make latexpdf` or `make -C <latex-build-dir> all-pdf` to build your project. The Sphinx contributed LaTeX makefile is adapted to the engine (lualatex) and language (German) and the LaTeX build repertory has specific xindy support files.

**Comment by funkyfuture** [comment]
no, i'm locally building with `make latexpdf`. i wouldn't share my project publicly (it' yet unpublished), but wouldn't mind to provide it to you. also, i wouldn't know how to use `xindy` directly, i just had that option set to `True`.

**Comment by jfbu** [comment]
@funkyfuture I understand you can't share your project, and on the other hand I will not contribute privately to your project, and I am not interested into spending myself the time to cut out from your project the irrelevant things. It is not even clear if the problem you have is  relevant to this issue or another one, and not seeing neither source code nor any image of output I can only try shots in the dark. If you have hit against a Sphinx bug relative to list of figures in PDF build (there is no such list of figures by default), then I will be very grateful if small project would reproduce it. For this we basically need conf.py, an index.rst, a few dummy PNG files and one or two paragraphs of source reST in one or two auxiliary files or a dummy Python module.

**Comment by funkyfuture** [comment]
@jfbu sorry, i didn't want to give the impression that i'm seeking personal support. i'll see that i may open issues at some point, but given that i usually don't require pdf output and considering the amount of open issues, i don't find it that important.

**Comment by jfbu** [comment]
@funkyfuture no problem, I was only a bit frustrated from sensing I don't have all details: you mentioned list of figures, but by default the Sphinx LaTeX templates makes no provision for including in PDF output a list of figures, so my feeling is you have some custom set-up which may explain the trouble with a custom References entry missing in the bookmarks/TOC.
